rosbag2_cpp/src/rosbag2_cpp/message_definitions/local_message_definition_source.cpp

```diff
@@ -256,9 +263,10 @@ rosbag2_storage::MessageDefinition LocalMessageDefinitionSource::get_full_text(
   Format format = Format::UNKNOWN;
   int32_t max_recursion_depth = ROSBAG2_CPP_LOCAL_MESSAGE_DEFINITION_SOURCE_MAX_RECURSION_DEPTH;
 
-  bool is_action_type = is_topic_belong_to_action(topic_name, root_type);
+  bool is_action_type = root_type.find("/action/") != std::string::npos;
```

Comment:
When handling action interface  `goal_cancel` and `status`, there are issues because regardless of the action message, their message types are fixed (action_msgs/srv/CancelGoal_Event and action_msgs/msg/GoalStatusArray).  

```suggestion
  bool is_action_type = root_type == "action_msgs/msg/GoalStatusArray"
     || root_type == "action_msgs/msg/GoalStatusArray"  
     || root_type.find("/action/") != std::string::npos;
```